integration/spark/shared/src/test/java/io/openlineage/spark/agent/lifecycle/plan/BuiltInLineageV1RelationHandlerTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineage.InputDataset;
+import io.openlineage.client.utils.DatasetIdentifier;
+import io.openlineage.spark.agent.Versions;
+import io.openlineage.spark.agent.lifecycle.plan.handlers.BuiltInLineageV1RelationHandler;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.builtin.scala.v1.LineageRelation;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.sources.BaseRelation;
+import org.junit.jupiter.api.Test;
+
+public class BuiltInLineageV1RelationHandlerTest {
```

Comment:
A general comment for test: would be good to have one

1. we define new both root/leaf `Fake` LogicalPlan node that implements provided interface
2. we create full logical plan containing those nodes
3. we use new mechanism to extract e2e information from them

Reply:
I agree that it would be good to have one. However, I don't know how to achieve this without writing an own Spark extension for testing purposes. I don't think this can be achieved with mocked/fake logical plans. 

If we were about to write custom Spark extension for testing purpose, I would recommend implementing traits in some real-life extension and including it later in our integration tests.

Reply:
Would be good as part of proposed Iceberg extension, however I agree it's outside of the scope of this (already very large) PR.